packages/flutter/lib/src/painting/text_painter.dart

```diff
@@ -239,13 +247,54 @@ class TextPainter {
       ); // direction doesn't matter, text is just a space
       if (text?.style != null)
         builder.pushStyle(text.style.getTextStyle(textScaleFactor: textScaleFactor));
+      // TODO(garyq): using an alphabetic space character is English bias, and should be
+      // removed and replaced with a less biased alternative. We can choose what
+      // character to use based of the locale and the ScriptCategory field of
+      // MaterialLocalization.
       builder.addText(' ');
```

Comment:
Hm. Really ideally we'd never change the caret size... it's a pity we don't have a way to just pick the "right" size for all text. Maybe we could measure the string U+0020 U+3000 ?

The locale from dart:ui isn't helpful here. We have no idea what the locale of the actual TextPainter's text is (it might have nothing to do with the locales in dart:ui).